File: 2788/go/main.go
package main

func splitWordsBySeparator(words []string, separator byte) []string {
	result := make([]string, 0)
	for _, word := range words {
		item := ""
		for _, c := range word {
			if byte(c) == separator {
				if len(item) > 0 {
					result = append(result, item)
					item = ""
				}
			} else {
				item += string(c)
			}
		}
		if len(item) > 0 {
			result = append(result, item)
			item = ""
		}
	}
	return result
}

File: 2788/python3/main.py
class Solution:
    def splitWordsBySeparator(self, words: list[str], separator: str) -> list[str]:
        result = []
        for word in words:
            sub = word.split(separator)
            for it in sub:
                if len(it) > 0:
                    result.append(it)

        return result
